### PR TITLE
build: handle non-git repo builds

### LINKED ISSUE #109: Building requires git repo
Users downloading the zip will get build failures since we directly reference git variables.

Here are some (might be others)
https://github.com/analogdevicesinc/iio-oscilloscope/blob/master/dialogs.c#L906
https://github.com/analogdevicesinc/iio-oscilloscope/blob/c37d0c2fa1e68dfdb301b4fef6cdc8974c91f102/dialogs.c#L946

### PR DESCRIPTION
**Pull request by commodo**
Fixes #109 

Signed-off-by: Alexandru Ardelean <alexandru.ardelean@analog.com>

### CODE REVIEW

File: dialogs.c [hunk 1]
```diff
@@ -900,10 +904,20 @@ gint create_blocking_popup(GtkMessageType type, GtkButtonsType button,
 	return run;
 }
 
+static inline glong date_build_time()
+{
+#ifdef GIT_COMMIT_TIMESTAMP
+	return atol(GIT_COMMIT_TIMESTAMP);
+#else
+	/* g_get_real_time() returns microseconds */
+	return (glong)(g_get_real_time() / 1000);
```
[the remark this answers is not in the document]
Comment (commodo) [reply dialogs.c]: ack

**Comment by commodo** [comment]
Changelog v1 -> v2:
* reworked commit `dialogs.c: handle empty GIT_COMMIT_TIMESTAMP & GIT_VERSION ` ; `date_compare_against_build_date()` returns 0 if `GIT_COMMIT_TIMESTAMP` is undefined

**Comment by commodo** [comment]
Funny thing is that it's only happening with cmake.
I didn't see it without cmake tho.

Thanks for catching this :)

Because I goofed with this, I owe an auto-test, so I added one.

So, changelog v1 -> v2:
* updated cmake build & make build to make sure no CFLAGS are added for non-git repos ; tested them manually
* added a Travis-CI build step to test the case of a non-git repo
  * [note] not testing the regular Makefile, as that one is intended for removal

**Comment by commodo** [comment]
Changelog v3 -> v4:
* tweaked Travis CI build with recent changes ; some things reproduce in CI only

**Comment by commodo** [comment]
Will merge this later today if no objections

**Comment by lclausen-adi** [comment]
Adding @rgetz to get some ideas on what should happen with the version check when not building from git.

**Comment by commodo** [comment]
ack
will defer merging this ;

my input on this:
* we could try to get rid of the old Makefile, to simplify building iio-osciloscope on Windows ; or to try to cover that as well
* in the CMakeLists.txt file, the current OSC_VERSION is set : https://github.com/analogdevicesinc/iio-oscilloscope/blob/master/CMakeLists.txt#L22  ; this is similar to what Scopy is doing atm, with keeping the version in CMakeLists.txt ; 
* changing the version scheme would also work ; i.e. not use GIT_VERSION  and use an OSC_VERSION from a Makefile / CMakeLists.txt ; not completely sure how that would work with respect to updates, but it would help to not depend on the git data for version info

**Comment by commodo** [comment]
@rgetz 
ping

If this gets no feedback by Friday [12th Oct] I'll probably merge it as is, and we can address other issues that come up later.

**Comment by commodo** [comment]
If there are any unresolved topics with respect to versioning in a non-git repo, we can address them in an upcoming patch-series.

**Pull request merged**